### DISCUSSION TITLE
"Unable to reflow"

### DISCUSSION THREAD

**Discussion by petrvokac** [pycalphad/pycalphad#407] [Q&A] [answered]
Hello,

I am trying to read tdb and write modified one. However I am getting "Unable to reflow" error. E.g.:

`from pycalphad import Database
db = Database('cost507R.TDB')
db.to_file('cost507RW.TDB')`

where cost507R.TDB is from OpenCalphad6 installation.

The output is:

`Traceback (most recent call last):
  File "/home/pet/Documents/calphad/pycalphad/examples/rw.py", line 3, in <module>
    db.to_file('cost507RW.TDB')
  File "/home/pet/.local/lib/python3.9/site-packages/pycalphad/io/database.py", line 303, in to_file
    format_registry[fmt].write(self, fd, **write_kwargs)
  File "/home/pet/.local/lib/python3.9/site-packages/pycalphad/io/tdb.py", line 827, in write_tdb
    fd.write(reflow_text(output, linewidth=maxlen))
  File "/home/pet/.local/lib/python3.9/site-packages/pycalphad/io/tdb.py", line 542, in reflow_text
    raise ValueError(f"Unable to reflow the following line of length {len(line)} below the maximum length of {linewidth}: \n{line}")
ValueError: Unable to reflow the following line of length 167 below the maximum length of 78: 
   4.005269*T-1081244.0*T**(-1)-0.016814429*T**2-1.2307E+25*T**(-9)-8.2590486*T*LOG(T); 800.0 Y -8256.959 + 138.99688*T-28.4512*T*LOG(T)-1.2307E+25*T**(-9); 3000.0 N !
`
Any hint?

Thank you.
Petr

**Comment by richardotis** [comment, marked as answer]
This is a bug with how pycalphad writes out long numerical expressions and then attempts to add line breaks to make it fit within 80 characters (which used to be a hard limit in TC). I think we've fixed this bug in the latest development version, but if you can share the TDB file I can confirm.

As a workaround until we confirm the bug is fixed and release a new version, try adding this to the top of your code to disable reflowing text:
```python
import pycalphad.io.tdb
pycalphad.io.tdb.reflow_text = lambda text, **kwargs: text
from pycalphad import Database
db = Database('cost507R.TDB')
db.to_file('cost507RW.TDB')
```
I'm not sure if recent versions of TC or OC will accept files with long lines, but pycalphad should be able to read it back.

**Reply by bocklund** [reply]
I tested the cost507R distributed with OpenCalphad and verified the ValueError for failling to reflow is triggered on 0.10.0 and fixed on `develop`.

**Reply by petrvokac** [reply]
Thank you,
I tried modified script as indicated and it seems to work (no errors). OC issues warning on lines longer than 80 but seems to work with it. In fact I am not interested in 80 characters limit but I want find easiest way to read and modify tdb. I am programming in Python, hence I consider Python API more easier for me than OCs' macros. Exercise with cost507R was just an example of the first trouble I experienced. Now I can go on with my experiments. I am also looking forward for the next pycalphad version.
Thank you. 
Petr
